src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,297 @@
+import {faChevronDown, faSearch} from "@fortawesome/pro-regular-svg-icons";
+
+export interface DropdownProps {
+  label?: string;
+  placeholder?: string;
+  options?: any;
+  loading?: boolean;
+  defaultValue?: React.ReactNode;
+  maxWidth?: string;
+  multipleSelect?: boolean;
+  hasSelectedCount?: boolean;
+  hasClearSelection?: boolean;
+  hasSearch?: boolean;
+  onChange?: (selectedValue: any, previouslySelected: boolean) => void;
+  onClearSelection?: () => void;
+  error?: string;
+  "data-testid"?: string;
+  "data-cy"?: string;
+  loadingIndicator?: React.ReactNode;
+  selectedIndicator?: React.ReactNode;
+}
+
+/**
+ *
+ * `Dropdown` Describe what it does
+ *
+ * Component Demo: [Dropdown](https://ui.govconnex.com/?path=/story/components-Dropdown--example)
+ *
+ */
+const Dropdown = ({
+  label,
+  placeholder,
+  options = [],
+  loading,
+  defaultValue,
+  maxWidth,
+  multipleSelect = false,
+  hasSelectedCount = false,
+  hasClearSelection = false,
+  hasSearch = false,
+  onChange = () => {},
+  onClearSelection = () => {},
+  error,
+  "data-testid": dataTestId,
+  "data-cy": dataCy,
+  loadingIndicator,
+  selectedIndicator,
+}: DropdownProps) => {
+  const popRef = useRef<any>(null);
+  const [showPopover, setShowPopover] = useState(false);
+  // eslint-disable-next-line
+  const [value, setValue] = useState(defaultValue);
+  const [searchTerm, setSearchTerm] = useState("");
+  const selectedCount = [
+    ...new Set(
+      options?.filter((item: any) => item?.selected)?.map((item: any) => item?.id)
+    ),
+  ]?.length;
+
+  // all is selected if there is nothing selected(for empty selected) and not selected is 0 (for all selected)
+  const notSelectedCount = [
+    ...new Set(
+      options?.filter((item: any) => !item?.selected)?.map((item: any) => item?.id)
+    ),
+  ]?.length;
+
+  const optionsDisplayed = searchTerm
+    ? (options || []).filter(
+        (item: any) => item?.text?.toLowerCase()?.includes(searchTerm?.toLowerCase())
+      )
+    : options || [];
+
+  return (
+    <ClickAwayListener
+      onClickAway={(e) => {
+        // Check if the clicked element is the anchorEl, if so, prevent the clickaway logic
+        if (popRef?.current && popRef?.current?.contains(e?.target)) {
+          return;
+        }
+
+        // Your regular clickaway logic here
+        setShowPopover(false);
+      }}
+    >
+      <Box>
+        <Input
+          data-testid={dataTestId}
+          data-cy={dataCy}
+          error={error}
+          onClick={() => {
+            setShowPopover(true);
+          }}
+          endAdornment={
+            <Box
+              cs={{
+                display: "inline-flex",
+                gap: "spacing.xs",
+                alignItems: "center",
+                paddingLeft: "spacing.sm",
+                background: "white",
+              }}
+            >
+              {hasSelectedCount ? (
+                !notSelectedCount ? (
+                  // eslint-disable-next-line
```

Comment:
Question; why are we disabling eslint here?

Reply:
eslint is looking for the aria attribute role. The chip component has a "role" property so this could end up being wrong. I could put a comment on it

Reply:
Okay, yeah we need to ensure we don't use standard attribute names for custom props in future.